app/renderer/ipc.ts

```diff
@@ -0,0 +1,21 @@
+import { ipcRenderer } from "electron"
+export * from "app/common/ipc"
+
+export type Listener = (e: Event, message: string) => void
+
+export type Ipc = {
+  once(channel: string, listener: Listener): void
+  send(channel: string, ...args: Array<any>): void
+  sendSync(channel: string, ...args: Array<any>): any
+  removeAllListeners(channel: string): void
+}
+
+export const electronIpc: Ipc = ipcRenderer
+
+export const idGenerator = (() => {
+  let current = 0
+
+  return () => current = (current + 1) % Number.MAX_SAFE_INTEGER
+})()
```

Comment:
I have my doubts about usage of this kind of stateful closures as they are hardly readable.

However, only other option that comes to my mind is usage of generators, which don't look that much better anyway, so never mind:
```ts
const idGenerator = (function* () {
  let current = 0
  while (true) {
    current = (current + 1) % Number.MAX_SAFE_INTEGER
    yield current
  }
})()
```

Reply:
Maybe a good ol' class is the best solution, what do you think?

Reply:
I think a class would have roughly the same API as the generator provided that both are defined as singletons:
```ts
const idGenerator = new class {
  private current = 0
  public next = () => {
    current = (current + 1) % Number.MAX_SAFE_INTEGER
    return current
  }
}

idGenerator.next() // 1
idGenerator.next() // 2
```
We can go with a class if you think it's more legible.